ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2255,17 +2255,19 @@ MiInitializeLoadedModuleList(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
         }
 
         /* Set the base name */
+        // NewEntry->BaseDllName.Length = LdrEntry->BaseDllName.Length; // Already.
+        NewEntry->BaseDllName.MaximumLength = NewEntry->BaseDllName.Length + sizeof(UNICODE_NULL);
         NewEntry->BaseDllName.Buffer = (PVOID)(NewEntry + 1);
 
-        /* Copy the full and base name */
+        /* Copy and null-terminate the full and base names */
```

Comment:
You seem to be determined to change as many lines in the kernel as possible. I really don't want git blame to one day tell me that you wrote the entire kernel.